example/e2e/deploy-federated-nginx.sh

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script has an end to end steps in deploying a federated deployment
+
+set -o nounset
+set -o pipefail
+
+function wait-for-conditions() {
+    local timeout=$1
+    local condition1=$2
+    local condition2=$3
+
+    local counter=0
+    local ret1=1
+    local ret2=1
+    while [[ ${counter} -lt ${timeout} ]]; do
+        if [[ ${ret1} -ne 0 ]]; then
+            eval "(${condition1}) &> /dev/null"
+            ret1=$?
+        fi
+        if [[ ${ret2} -ne 0 ]]; then
+            eval "(${condition2}) &> /dev/null"
+            ret2=$?
+        fi
+
+        if [[ ${ret1} -eq 0 && ${ret2} -eq 0 ]]; then
+            echo
+            return 0
+        fi
+
+        counter=$((counter + 1))
+        sleep 1
+        echo -n "."
+    done
+
+    echo
+    return 1
+}
+
+TEST_NS="test-namespace"
+
+# Prerequisite:
+# - a cluster scoped federation deployment with `cluster1` and
+#   `cluster2` joined as member clusters
+# - # kubectl config use-context cluster1
+#
+echo "Starting deploying a ngnix application in federation cluster."
+
+echo
+echo "Creating a test namespace ${TEST_NS}."
+kubectl create ns ${TEST_NS}
+if [[ $? != 0 ]]; then
+    echo "Failed creating ${TEST_NS}."
+    exit 1
+fi 
+
+echo
+echo "Creating a ngnix deployment in namespace ${TEST_NS}."
+kubectl apply -n ${TEST_NS} -f yaml/configmap.yaml
+if [[ $? != 0 ]]; then
+    echo "Failed creating a configmap for the ngnix application."
+    exit 1
+fi 
+kubectl apply -n ${TEST_NS} -f yaml/deployment.yaml
+if [[ $? != 0 ]]; then
+    echo "Failed creating a ngnix deployment."
+    exit 1
+fi 
+
+echo
+echo "Creating a ngnix service in namespace ${TEST_NS}."
+kubectl apply -n ${TEST_NS} -f yaml/service.yaml
+if [[ $? != 0 ]]; then
+    echo "Failed creating a ngnix service."
+    exit 1
+fi 
+
+echo
+echo "Federating resources in namespace ${TEST_NS} to member clusters."
+kubefedctl federate ns ${TEST_NS} --contents --skip-api-resources 'pods,secrets,serviceaccount,replicaset'
+if [[ $? != 0 ]]; then
+    echo "Failed federating resources to member clusters."
+    exit 1
+fi 
+
+
+echo
+echo "Checking status of federated resources."
```

Comment:
There is no point in retrieving status if it won't be evaluated.  Either the result of a jsonpath query should be evaluated or this check should be skipped.